### PR TITLE
Peel off outer ForwardingStatement after running semantic

### PR DESCRIPTION
**Pull request by ibuclaw**
FYI - @tgehr 

As it has no use outside of semantic analysis.

### CODE REVIEW

**Comment by ibuclaw** [comment]
This should be ready for review. @tgehr

File: src/dmd/blockexit.d [hunk 1]
```diff
@@ -190,14 +190,7 @@ int blockExit(Statement s, FuncDeclaration func, bool mustNotThrow)
...
         {
-            if (s.statement)
-            {
-                s.statement.accept(this);
-            }
-            else
-            {
-                result = BE.fallthru;
-            }
+            assert(0);
```
Comment (tgehr) [inline src/dmd/blockexit.d]: I guess this can be removed, as the Statement overload will assert anyway.

Comment (tgehr) [inline src/dmd/blockexit.d]: Looks good to me. Thanks for taking this!

**Review by tgehr — APPROVED** [review]

**Comment by tgehr** [comment]
Looks good to me. Thanks for taking this!

**Review by PetarKirov — APPROVED** [review]

**Pull request merged**